src/server/routes/contract/extensions/erc1155/write/transfer.ts

```diff
@@ -67,24 +76,38 @@ export async function erc1155transfer(fastify: FastifyInstance) {
         "x-backend-wallet-address": walletAddress,
         "x-account-address": accountAddress,
         "x-idempotency-key": idempotencyKey,
+        "x-account-factory-address": accountFactoryAddress,
+        "x-account-salt": accountSalt,
       } = request.headers as Static<typeof walletWithAAHeaderSchema>;
 
       const chainId = await getChainIdFromChain(chain);
       const contract = await getContract({
-        chainId,
-        contractAddress,
-        walletAddress,
-        accountAddress,
+        client: thirdwebClient,
+        chain: await getChain(chainId),
+        address: contractAddress,
       });
-      const tx = await contract.erc1155.transfer.prepare(to, tokenId, amount);
 
-      const queueId = await queueTx({
-        tx,
-        chainId,
-        simulateTx,
-        extension: "erc1155",
-        idempotencyKey,
+      const transaction = safeTransferFrom({
+        contract,
+        from: getChecksumAddress(walletAddress),
+        to: getChecksumAddress(to),
+        tokenId: BigInt(tokenId),
+        value: BigInt(amount),
+        data: "0x",
+      });
+
+      const queueId = await queueTransaction({
+        transaction,
+        fromAddress: getChecksumAddress(walletAddress),
+        toAddress: getChecksumAddress(contractAddress),
```

Comment:
Making sure the from/to here is always backend wallet -> contract (as opposed to above when building the transaction where from/to are the sender/recipient addresses).